sound/soc/sof/ipc.c

```diff
@@ -306,6 +306,27 @@ int sof_ipc_tx_message(struct snd_sof_ipc *ipc, u32 header,
 }
 EXPORT_SYMBOL(sof_ipc_tx_message);
 
+/* send IPC message from client to DSP */
+int sof_client_tx_message(struct device *dev, u32 header,
+			  void *msg_data, size_t msg_bytes, void *reply_data,
+			  size_t reply_bytes)
+{
+	struct snd_sof_client *sof_client = dev_get_platdata(dev);
+	struct snd_sof_dev *sdev = dev_get_drvdata(dev->parent);
+	int ret;
+
+	ret = sof_ipc_tx_message(sdev->ipc, header, msg_data, msg_bytes,
+				 reply_data, reply_bytes);
+
+	/* invoke client reply callback if IPC is successful */
+	if (sof_client->sof_ipc_reply_cb)
+		sof_client->sof_ipc_reply_cb(sof_client, reply_data,
```

Comment:
@lyakh the IPC TX request could be coming from any of the clients and the callback is just a way to return the response to the respective client. In the current case, the clients dont really care for the reply but that could change in the future.

Reply:
@ranj063 to the calling client, right? Not to some pre-registered entity, that wants to be notified every time an IPC message is sent? Then why just return the result in a function parameter isn't enough? Especially if you don't have any current use-cases, I wouldn't put this there "just in case"

Reply:
@lyakh yes OK makes sense. Let me drop the TX callback.